code/modules/modular_computers/file_system/programs/generic/folding.dm

```diff
@@ -0,0 +1,119 @@
+GLOBAL_LIST_INIT(science_strings, list(
+	"Extruding Mesh Terrain", "Virtualizing Microprocessor",
+	"Reticulating Splines", "Inserting Chaos Generator",
+	"Reversing Polarity", "Unfolding Proteins",
+	"Simulating Alien Abductions", "Scanning Pigeons",
+	"Iterating Chaos Array", "Abstracting Supermatter",
+	"Adjusting Social Network", "Recalculating Clown Principle"
+))
+
+#define MINIMUM_SCIENCE_INTERVAL 450
+#define MAXIMUM_SCIENCE_INTERVAL 900
+#define MINIMUM_FOLDING_EVENT_INTERVAL 50
+#define MAXIMUM_FOLDING_EVENT_INTERVAL 100
+#define SCIENCE_MONEY_PER_MINUTE 0.08 // So little money.
+
+/datum/computer_file/program/folding
+	filename = "fldng"
+	filedesc = "FOLDING@SAPCE"
+	extended_desc = "This program uses processor cycles for science, in exchange for money."
+	program_icon_state = "generic"
+	program_key_state = "generic_key"
+	size = 6
+	available_on_network = 1
+	usage_flags = PROGRAM_ALL
+	nanomodule_path = /datum/nano_module/program/folding
+	category = PROG_UTIL
+	requires_network = 1
+
+	var/started_on 			= 0 	// When the program started some science.
+	var/current_interval 	= 0		// How long the current interval will be.
+	var/next_event 			= 0 	// in world timeofday, when the next event is scheduled to pop.
+	var/crashed				= FALSE	// Program periodically needs a restart.
+	var/crashed_at			= 0		// When the program crashed.
+
+/datum/computer_file/program/folding/Topic(href, href_list)
+	if(..())
+		return TOPIC_HANDLED
+	. = TOPIC_REFRESH
+
+	if(href_list["fix_crash"] && crashed)
+		current_interval += world.timeofday - crashed_at
+		crashed = FALSE
+
+	if(href_list["start"] && started_on == 0)
+		started_on = world.timeofday
+		current_interval = rand(MINIMUM_SCIENCE_INTERVAL, MAXIMUM_SCIENCE_INTERVAL) SECONDS
+		// Better processors can do this faster. Normal processors are slower.
+		var/obj/item/stock_parts/computer/processor_unit/processor = computer.get_component(PART_CPU)
+		current_interval = (processor.processing_power / 2) * current_interval
+		next_event = (rand(MINIMUM_FOLDING_EVENT_INTERVAL, MAXIMUM_FOLDING_EVENT_INTERVAL) SECONDS) + world.timeofday
+
+	if(href_list["collect"] && started_on > 0 && !crashed)
+		var/obj/item/card/id/I = usr.GetIdCard()
+		if(!I)
+			to_chat(usr, SPAN_WARNING("Unable to locate ID card for transaction."))
+			return TOPIC_HANDLED
+		var/datum/money_account/account = get_account(I.associated_account_number)
+		var/earned = current_interval * SCIENCE_MONEY_PER_MINUTE
```

Comment:
Making it based directly on `current_interval` means that more processing power doesn't actually make you more money. Should probably average the min/max science interval